python/cuml/neighbors/nearest_neighbors.pyx

```diff
@@ -525,6 +567,36 @@ class NearestNeighbors(Base):
         out_type = _output_type \
             if _output_type is not None else self._get_output_type(X)
 
+        if two_pass_precision:
+            metric, expanded = self._build_metric_type(self.metric)
+            metric_is_l2_based = (
+                metric == MetricType.METRIC_L2 or
+                (metric == MetricType.METRIC_Lp and self.p == 2)
+            )
+
+            # FAISS employs imprecise distance algorithm only for L2-based
+            # metrics
+            if metric_is_l2_based:
+                X = input_to_cupy_array(X).array
+                I_cparr = I_ndarr.to_output('cupy')
+
+                precise_distances = cp.linalg.norm(
```

Comment:
Why use linalg norm in the expanded case? Maybe just use square distance then? Seems like otherwise in the precise case we're doing extra work and possibly losing precision for expanded.

Reply:
No good reason! I think this was a holdover from when I thought I needed to do something a little more general. Cleaning it up now...

Reply:
Done! Good catch.